File: leetcode answers/1 - 10 leetcode solutions/6. Zigzag Conversion/Zigzag Conversion.py
class Solution:
    def convert(self,s,numRows):
        if numRows==1 or numRows>=len(s):
            return s
        rows=[""]*numRows
        row=0
        down=True
        for c in s:
            rows[row]+=c
            if row==0:
                down=True
            elif row==numRows-1:
                down=False
            row+=1 if down else -1
        return "".join(rows)

File: leetcode answers/1 - 10 leetcode solutions/6. Zigzag Conversion/Zigzag Conversion.js
var convert = function(s,numRows) {
    if(numRows===1||numRows>=s.length) return s;
    let rows=Array(numRows).fill("");
    let row=0,down=true;
    for(let c of s){
        rows[row]+=c;
        if(row===0) down=true;
        else if(row===numRows-1) down=false;
        row+=down?1:-1;
    }
    return rows.join("");
};